pkg/domain/scan.go

```diff
@@ -28,13 +28,32 @@ type Scan struct {
 	HostsStatus  []StatusHost    `json:"hosts_status,omitempty"`
 	HostsResults []ResultHost    `json:"hosts_results,omitempty"`
 	Targets      []events.Target `json:"targets,omitempty"`
+	CreatedAt    time.Time       `json:"created_at,omitempty"`
+	UpdatedAt    time.Time       `json:"updated_at,omitempty"`
 	StartedAt    time.Time       `json:"started_at"`
 	EndedAt      time.Time       `json:"ended_at"`
+	Status       string          `json:"status,omitempty"`
+}
+
+type ScanSummary struct {
```

Comment:
This struct clears things up much more! Just a question: I'm assuming the `Severity` attribute is this.

![image](https://github.com/user-attachments/assets/21267198-6200-4c9c-9109-0004bf4f2346)
 
In which case, this attribute being an array of integers is not very descriptive, and requires external context to interpret (e.g., which index represents "CRITICAL," "HIGH," etc.)

I suggest the following approach, to make our backend lives easier when it comes to filling out these attributes later on:

```suggestion
type SeverityCounts struct {
	Critical int `json:"critical"`
	High     int `json:"high"`
	Medium   int `json:"medium"`
	Low      int `json:"low"`
}


type ScanSummary struct {
ScanDate      string         `json:"scan_date,omitempty"`
	Host          string         `json:"host,omitempty"`
	Vulnerabilities int          `json:"vulnerabilities,omitempty"`
	Severities    SeverityCounts `json:"severities,omitempty"`
	Duration      int            `json:"duration,omitempty"`
	Status        int            `json:"status,omitempty"`
}

```

Reply:
Good approach for severities